doc/PORTABLE_SERVICES.md

```diff
@@ -0,0 +1,119 @@
+distribution images carry anyway.
+
+If you so will, "Portable Services" are a nicer way to manage chroot()
+environments, with better security, tooling and behaviour.
+
+# Where's the different to a "Container"?
+
+"Container" is a very vague term, after all it is used for
+systemd-nspawn/LXC-type OS containers, for Docker/rkt-like micro service
+containers, and even certain 'lightweight' VM runtimes.
+
+The "portable service" concept ultimately will not provide a fully detached
+environment to the payload, like containers intend to. Instead they are
+primarily much like regular system services, can be controlled with the same
+tools, are exposed the same way in all tools and so on. Their main difference
+is that the use a different root directory than the rest of the system. Hence,
+the intention is not to run code in a different, isolated world from the host —
+like most containers would do it —, but to run it in the same world, but with
+stricter access controls on what the service can see and do.
+
+As one point of differentiation: as programs run as "portable services" are
+pretty much regular system services, they won't run as PID 1 (like Docker would
+do it), but as normal process. A corrolary of that is that they aren't supposed
+to manage anything in their own system, such as the network, which is shared
+with the rest of the system.
+
+The primary focus use case of "portable services" is to extend the host system
+with encapsulated extensions, but provide almost full integration with the rest
+of the system, though possibly restricted by effective security knobs. This
+focus includes system extensions otherwise often called "super-privileged
+containers".
+
+Note that portable services are only available for system services, not for
+user services. i.e. the functionality cannot be used for the stuff
+bubblewrap/flatpak is focussing on.
+
+# Mode of Operation
+
+If you have portable service image, maybe in a raw disk image called
+`foobar.raw`, then attaching the services to the host is as easy as:
+
+```
+# /usr/lib/systemd/portablectl attach foobar.raw
+```
+
+This command does the following:
+
+1. It dissects the image, checks and validates the `/etc/os-release` data of
+   the image, and looks for all included unit files.
+
+2. It copies out all unit files with a suffix of `.service`, `.socket`,
+   `.target` and `.path` whose name begins with the image's name (with the .raw
+   removed), truncated at the first underscore (if there is one). This prefix
+   name generated from the image name must be followed by a "." or "-"
+   character in the unit name. Or in other words, given the image name of
+   `foobar.raw` all unit files matching `foobar-*.{service|socket|target|path}`
+   as well as `foobar.*.{service|socket|target|path}` and
+   `foobar.{service|socket|target|path}` are copied out. These unit files are
+   placed in `/etc/systemd/system/` like regular unit files.
```

Comment:
not /run/systemd/system ? that would make more sense to me...

Reply:
There's `--runtime` which you can specify on the `portablectl` cmdline. If you do the stuff is attached only until the next reboot, i.e. the files are copied to /run. If you don't the stuff s attached for good, i.e. to /etc.